app/models/patient_session.rb

```diff
@@ -126,4 +128,23 @@ def latest_triage
   def consents_to_send_communication
     latest_consents.select(&:response_given?).reject(&:via_self_consent?)
   end
+
+  def pending_transfer?
+    proposed_session_id.present?
+  end
+
+  def confirm_transfer!
+    return unless pending_transfer?
+
+    if vaccination_records.any?
+      PatientSession.create!(patient:, session: proposed_session)
+      update!(proposed_session: nil)
+    else
+      update!(session: proposed_session, proposed_session: nil)
```

Comment:
I'm wondering whether it's better to always create a new one and delete old ones? It means `created_at` can be relied on to mean when the patient was added to the session.

Reply:
Hmm, yeah that's a good idea

Reply:
Updated!